### DISCUSSION TITLE
How to achieve conditional slot?

### DISCUSSION THREAD

**Discussion by donPuerto** [quasarframework/quasar#10044] [General - Components / Directives / etc] [answered]
```
<template v-slot:prepend>
    <q-icon name="event" />
</template>
```
I want  prepend and append to be dynamic. How to achieve this?

**Comment by hawkeye64** [comment, marked as answer]
Put your `v-if="the-condition"` on your template declaration.

**Reply by donPuerto** [reply]
No i mean like this
```
<template v-slot:`${dynamic}`>
    <q-icon name="event" />
</template>

```
dynamic would be "prepend" and "append"
But this wont work of course. Any other way? @hawkeye64

**Reply by ontwikkelfabriek** [reply]
@donPuerto 

> dynamic would be "prepend" and "append" But this wont work of course. 

Why not? See this stackoverflow discussion about dynamic slots:
https://stackoverflow.com/questions/57156543/how-to-declare-a-list-of-slots-with-dynamic-names-and-dynamic-components-in-a-vu

**Comment by hawkeye64** [comment]
```js
<template v-slot:[dynamic]>
    <q-icon name="event" />
</template>
```

or

```js
<template v-if="dynamic === 'prepend' v-slot:prepend>
    <q-icon name="event" />
</template>
<template v-if="dynamic === 'append'" v-slot:append>
    <q-icon name="event" />
</template>
```

**Reply by donPuerto** [reply]
Oh i got your point, thanks
